mp4/file.go

```diff
@@ -42,6 +43,16 @@ const (
 	EncModeBoxTree = EncFragFileMode(1) // Encode all boxes in file tree
 )
 
+type DecFileMode byte
+
+const (
+	// DecModeLazyMdat doesn't not read Mdat data into memory,
+	// which means the decoding process requires less memory and faster.
+	DecModeLazyMdat DecFileMode = iota
+	// DecModeNormal read Mdat data into memory during file decoding.
+	DecModeNormal
+)
```

Comment:
The DecModeNormal must be default, since many tests and external programs are broken if not. You could set the default in NewFile, but I think it is better to just reorder the values.

Reply:
updated accordingly